google/resource_container_cluster.go

```diff
@@ -76,6 +75,30 @@ func resourceContainerCluster() *schema.Resource {
 				},
 			},
 
+			"master_authorized_networks_config": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:     schema.TypeBool,
+							Required: true,
```

Comment:
I wasn't entirely sure what made sense here for `Optional` / `Computed` / `Required` / `Default`.  In the end, `Required: true` made the most sense to me, but only marginally so.  I'll test upstream for implicit requirements/defaults tomorrow.

Reply:
I'd actually recommend not having an `enabled` field at all, and instead just relying on the presence of any `master_authorized_networks_config`s to determine what that value should be, similar to how we do autoscaling. The reason for this is that when `enabled` is false, GCP won't return anything for `cidrBlocks`:
```
"masterAuthorizedNetworksConfig": {},
```

This means that a config like this:
```
  master_authorized_networks_config {
    enabled     = false
    cidr_blocks = ["172.16.0.0/12", "10.0.0.0/8"]
  }
```
is equivalent to:
```
  master_authorized_networks_config {
    enabled     = false
  }
```
Because we want to set the state based on what's actually true in GCP, the state will always resemble the second config above, and thus planning immediately after applying with the first config will show a diff.

"But wait!" you might ask, "I added a test for the `enabled=false` case and it passed!" Ah yes, and this tripped me up for a while too. I'll explain in a comment below.

Reply:
Totally, I almost did that, but wasn't sure and opted for mimicking the API.  I'll follow that pattern elsewhere.